File: src/backend/api/enums.py
from enum import StrEnum, auto


class ProjectStatus(StrEnum):
    AWAITING = auto()
    IN_PROGRESS = auto()
    COMPLETED = auto()


class DocumentStatus(StrEnum):
    PENDING = auto()
    PROCESSED = auto()
    FAILED = auto()


class PromptLanguage(StrEnum):
    CATALONIAN = auto()
    CASTILLIAN = auto()
    ENGLISH = auto()


class FileType(StrEnum):
    ORIGINAL = auto()
    REDACTED = auto()


class ScopeType(StrEnum):
    PROJECT = auto()
    DOCUMENT = auto()


class CommitState(StrEnum):
    STAGED_CREATION = auto()
    STAGED_EDITION = auto()
    STAGED_DELETION = auto()
    COMMITTED = auto()


class AnchorOption(StrEnum):
    NW = auto()
    NE = auto()
    SE = auto()
    SW = auto()
    ZH = auto()

File: src/frontend/src/types/enums.ts
import { z } from 'zod';

// Enums - matching backend exactly
export const ProjectStatusEnumSchema = z.enum(['awaiting', 'in_progress', 'completed']);
export const DocumentStatusEnumSchema = z.enum(['pending', 'processed', 'failed']);
export const PromptLanguageEnumSchema = z.enum(['catalonian', 'castillian', 'english']);
export const FileTypeEnumSchema = z.enum(['original', 'redacted']);
// New enums aligned to backend
export const ScopeTypeEnumSchema = z.enum(['project', 'document']);
export const CommitStateEnumSchema = z.enum(['staged_creation', 'staged_edition', 'staged_deletion', 'committed']);

// Types
export type ProjectStatusEnum = z.infer<typeof ProjectStatusEnumSchema>;
export type DocumentStatusEnum = z.infer<typeof DocumentStatusEnumSchema>;
export type PromptLanguageEnum = z.infer<typeof PromptLanguageEnumSchema>;
export type FileTypeEnum = z.infer<typeof FileTypeEnumSchema>;
export type ScopeTypeEnum = z.infer<typeof ScopeTypeEnumSchema>;
export type CommitStateEnum = z.infer<typeof CommitStateEnumSchema>;
